rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -405,9 +405,27 @@ rmw_context_impl_s::configure_security(DDS_DomainParticipantQos * const qos)
   static const char * const uri_prefix = "";
 #endif /* !RMW_CONNEXT_DDS_API_PRO_LEGACY */
 
-  std::unordered_map<std::string, std::string> security_files;
-  if (!rmw_dds_common::get_security_files(
-      uri_prefix, this->base->options.security_options.security_root_path, security_files))
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rcutils_string_map_t security_files = rcutils_get_zero_initialized_string_map();
+  rcutils_ret_t ret = rcutils_string_map_init(&security_files, 0, allocator);
+
+  auto scope_exit_ws = rcpputils::make_scope_exit(
+    [&security_files]()
+    {
+      rcutils_ret_t ret = rcutils_string_map_fini(&security_files);
+      if (ret != RMW_RET_OK) {
+        RMW_SET_ERROR_MSG("Failed to fini string map for security");
+      }
+    });
+
+  if (ret != RMW_RET_OK) {
+    RMW_SET_ERROR_MSG("Failed to initialize string map for security");
+    return RMW_RET_ERROR;
+  }
```

Comment:
according to the implementation, `rcutils_string_map_fini` only needs to be called when `rcutils_string_map_init` succeeds? (rclcpp handles this way, but rcl always calls `rcutils_string_map_fini`, that i think i can fix with another PR.)

```suggestion
  rcutils_ret_t ret = rcutils_string_map_init(&security_files, 0, allocator);

  if (ret != RMW_RET_OK) {
    RMW_SET_ERROR_MSG("Failed to initialize string map for security");
    return RMW_RET_ERROR;
  }

  auto scope_exit_ws = rcpputils::make_scope_exit(
    [&security_files]()
    {
      rcutils_ret_t ret = rcutils_string_map_fini(&security_files);
      if (ret != RMW_RET_OK) {
        RMW_SET_ERROR_MSG("Failed to fini string map for security");
      }
    });
```